manifests/charts/gateways/istio-ingress/values.yaml

```diff
@@ -7,6 +7,7 @@ gateways:
     labels:
       app: istio-ingressgateway
       istio: ingressgateway
+      istio.io/rev: latest
```

Comment:
how does this get overridden?

Reply:
Also, I am pretty sure this is backwards incompatible as it will change the selector on deployment which is incompatible

Reply:
the chart needs to use the global.revision variable. I just published my PR prematurely so @esnible could take a look from a UX perspective.

Agree there may be upgrade problems with the existing work. I will manually test the upgrades before removing the do not merge label.

Reply:
I think there are only two options:
* Change the deployment name
* Do not change the selector

Right now the names are never changing so it seems like it might override each other, so maybe the first fix is to address the naming.

The actual steps the user will take are not 100% clear to me though so hard to give concrete advice. The description shows what will happen to the k8s objects but not how the user triggers that.

Reply:
Right. In the scenario I have mapped out, I have the concept of changing names, with a new label/selector. This way you can rotate ingress gateways from an older version of the proxy to a newer version. If we wish to restore the old name, I could rotate them twice. (one of the rotations would be temporary, much as the idea of the canary control plane name is meant to be temporary.

I am open to ideas on the exact mechanics of ingress upgrade though and plan to write an RFC prior to t he env wg meeting to submit to the istio/enhancements repository.

Reply:
John, agree the names must change. The RFC reflects this model - which ostensibly you designed in the multiple control planes design document. I am unclear on backward compatibility on label changes, although I believe, but have not prototyped, a label change is required on both the service and deployment.

I'll publish the RFC this morning - perhaps you could take a quick review if you have time. If not, hope to see you at the Env wg meeting today for your feedback.

Cheers,
-steve